Client/Frontend/Login Management/LoginListViewController.swift

```diff
@@ -166,6 +176,23 @@ class LoginListViewController: UIViewController {
     }
 }
 
+// MARK: - Favicon Loading
+extension LoginListViewController {
+
+    private func loadFaviconsForLogins(logins: [Login]) -> Deferred<Maybe<[GUID: [Favicon]]>> {
+        let urls = logins.map { $0.hostname }
+        return self.profile.favicons.getFaviconsForHistoryURLs(urls) >>== { favicons in
+            let filteredFavicons = favicons.asArray().flatMap { $0 }
```

Comment:
If you make `getFaviconsForHistoryURLs` not return nulls, you can probably fight the type system to avoid this `flatMap`.